src/utils/hooks.js

```diff
@@ -19,3 +24,92 @@ export function useStateWithStorage(storageKey, initialValue) {
 
 	return [value, setValue];
 }
+
+export function useUrgency(items) {
+	const [urgencyObject, setUrgecyObject] = useState({
+		overdue: [],
+		soon: [],
+		kindOfSoon: [],
+		notSoon: [],
+		inactive: [],
+	});
+
+	const getUrgency = (name) => {
+		const statusArray = Object.entries(urgencyObject).find(
+			([status, items]) => {
+				if (items.some((item) => item.name === name)) {
+					return status;
+				}
+			},
+		);
+		if (!statusArray) {
+			throw new Error(`Failed to get class name of ${name}`);
+		}
+		return statusArray[0];
+	};
+
+	const sortByUrgency = (item, daysUntilNextPurchase) => {
+		if (daysUntilNextPurchase < 0) {
+			return 'overdue';
+		} else if (daysUntilNextPurchase < 7) {
+			return 'soon';
+		} else if (daysUntilNextPurchase >= 7 && daysUntilNextPurchase < 30) {
+			return 'kindOfSoon';
+		} else if (daysUntilNextPurchase >= 30) {
+			return 'notSoon';
+		} else {
+			throw new Error(`Failed to place [${item.name}]`);
+		}
+	};
```

Comment:
Could we create a `urgencyUtils.js` and move this urgency-related function to this new file?